### DISCUSSION TITLE
vweb: `return error("A bad request message")` be equal to `app.set_status(400, '') return app.text('error: ${err}')`

### DISCUSSION THREAD

**Discussion by enghitalo** [vlang/v#18626] [Ideas]
### Describe the feature

```v
module main

import vweb

struct App {
	vweb.Context
}

fn main() {
	mut app := &App{}
	vweb.run(app, 8082)
}

['/controller/user/create'; post]
pub fn (mut app App) controller_create_user(username string, password string) !vweb.Result {
	if username == '' { return error("A bad request message") } // This may return a status == 400, body == "A bad request message"

	// Like this
	app.set_status(400, '')
	return app.text('error: ${err}')
}
```

### Use Case

NestJS already do something like this

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.2 b8571c9

### Environment details (OS name and version, etc.)

```bash
OS: linux, Ubuntu 22.10
Processor: 16 cpus, 64bit, little endian, AMD Ryzen 7 5800H with Radeon Graphics
CC version: cc (Ubuntu 12.2.0-3ubuntu1) 12.2.0

getwd: /home/hitalo/Documents/v
vmodules: /home/hitalo/.vmodules
vroot: /home/hitalo/v
vexe: /home/hitalo/v/v
vexe mtime: 2023-01-04 11:05:35
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 458e68e.b8571c9

Git version: git version 2.37.2
Git vroot status: weekly.2023.01-14-gb8571c96 (1 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```

**Comment by JalonSolov** [comment]
The main problem with that is that the error may not be a 400.  Maybe it's a 400, maybe 401, maybe 403, etc.

**Comment by enghitalo** [comment]
> The main problem with that is that the error may not be a 400. Maybe it's a 400, maybe 401, maybe 403, etc.

Maybe 
`if username == '' { return http.error("A bad request message",401) }`?
or
`if username == '' { return vweb.error("A bad request message",401) }`?

**Comment by JalonSolov** [comment]
Yes, a way to specify the error code would be fine, just not automatic 400.

Perhaps we just need an `app.error()` to match `app.text()`.
